src/SourceBuild/content/repo-projects/arcade.proj

```diff
@@ -3,20 +3,23 @@
   <PropertyGroup>
     <!-- Use a prebuilt Arcade to build Arcade -->
     <UseBootstrapArcade>true</UseBootstrapArcade>
+    <ExtractToolPackageDependsOn>_DiscoverToolPackages</ExtractToolPackageDependsOn>
   </PropertyGroup>
 
   <ItemGroup>
     <RepositoryReference Include="source-build-reference-packages" Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
   </ItemGroup>
 
-  <ItemGroup>
-    <BuiltSdkPackageOverride Include="Microsoft.DotNet.Arcade.Sdk" Version="$(OutputPackageVersion)" />
-    <BuiltSdkPackageOverride Include="Microsoft.DotNet.SharedFramework.Sdk" Version="$(OutputPackageVersion)" />
-    <BuiltSdkPackageOverride Include="Microsoft.DotNet.CMake.Sdk" Version="$(OutputPackageVersion)" />
-  </ItemGroup>
-
   <ItemGroup>
     <ExtraPackageVersionPropsPackageInfo Include="NuGetVersion" Version="%24(NuGetPackagingVersion)" />
   </ItemGroup>
 
+  <Target Name="_DiscoverToolPackages" 
+          DependsOnTargets="RepoBuild;GetProducedPackages">
+    <ItemGroup>
+      <BuiltSdkPackageOverride Include="@(ProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.Arcade.Sdk'))" />
+      <BuiltSdkPackageOverride Include="@(ProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.SharedFramework.Sdk'))" />
+      <BuiltSdkPackageOverride Include="@(ProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.CMake.Sdk'))" />
+    </ItemGroup>
+  </Target>
```

Comment:
I'm a friend of doing as much work as possible statically without targets as it makes a project file more readable.

Could this be an item group as it was before (without setting a version) and make a target in Directory.Build.targets then read these items and set the version accordingly?

Reply:
We'd end up needing to use the `Join` task from the .NET SDK to get anywhere near the amount of clarity we have with the current setup. I'll look into using that.